src/main/java/io/usethesource/vallang/INode.java

```diff
@@ -30,6 +30,14 @@
  * it recursively.
  */
 public interface INode extends IValue, Iterable<IValue> {
+
+	@Override
+	default int getMatchFingerprint() {
+		int hash = getName().hashCode();
+
+		return hash == 0 ? 13547528 /* "node".hashCode() << 2*/ + arity() : hash + 131 * arity(); 
```

Comment:
I would like to propose to simplify this a bit

```
if (hash == 0) {
   hash = 13547528 /* "node".hashCode() << 2*/;
}
return hash + 131 * arity();
```

Reply:
ok. it's only one multiplication more expensive for the empty string case.